### ISSUE TITLE
no such file @ rb_file_size_s

### ISSUE THREAD

**Issue by obromios** [brigade/overcommit#571]
I had used overcommit for a particular repository, but had removed the gem, but  then doing a commit I received the following message
```
This repository contains hooks installed by Overcommit, but the `overcommit` gem is not installed.
Install it with `gem install overcommit`.
```
To fix this I rebooted my machine and, but when I tried to do a commit, I received the following message:

```
No such file or directory @ rb_file_s_size - /Users/Chris/Sites/philomaths/--git-common-dir/hooks/overcommit-hook
/Users/Chris/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:815:in `size'
/Users/Chris/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/fileutils.rb:815:in `compare_file'
/Users/Chris/.rvm/gems/ruby-2.2.1@philomaths_221/gems/overcommit-0.44.0/lib/overcommit/installer.rb:57:in `update'
/Users/Chris/.rvm/gems/ruby-2.2.1@philomaths_221/gems/overcommit-0.44.0/lib/overcommit/installer.rb:20:in `run'
.git/hooks/pre-commit:67:in `<main>'
Report this bug at https://github.com/brigade/overcommit/issues
```
so I am reporting it. 
Another aspect of the problem is that if do ```gem install overcommit``` and then do ```overcommit --uninstall``` following by ```overcommit --list-hooks``` the hooks are still there including some of the PreCommit hooks e.g. AuthorEmail being enabled.  If I then try to do a commit, I get the original error message i.e.
```
This repository contains hooks installed by Overcommit, but the `overcommit` gem is not installed.
Install it with `gem install overcommit`.
```
 Note that I am using RVM, but I have also tried ```rvm @global do gem uninstall overcommit```, which is appears to work,  but this does fix the issue of not being able to do a git commit.  I have also asked this [SO question ](https://stackoverflow.com/questions/49649015/overcommit-gem-is-preventing-a-commit) but to no avail.

**Comment by sds** [comment]
Hey @obromios, removing the gem does *not* remove the hooks. Remember that Git hooks are actual files installed in your repository's `.git/hooks` directory.

Rather than uninstall the gem, you needed to run:

```bash
overcommit --uninstall
```

...in your repository to uninstall the hooks. Hope that helps!

**Comment by obromios** [comment]
I am in the root of my repository at ```/Users/Chris/Sites/philomaths```,  ```ls -a``` shows Gemfile, .git etc. I type 
`overcommit -u`
and see 
`Successfully removed hooks from /Users/Chris/Sites/philomaths`
but when I do a commit, I still get 
```
This repository contains hooks installed by Overcommit, but the `overcommit` gem is not installed.
Install it with `gem install overcommit`.
```
and if I type ```overcommit --list-hooks``` I get a long list of commit hooks starting with
```
CommitMsg:
  CapitalizedSubject: enabled
  EmptyMessage: enabled
  GerritChangeId: disabled
  HardTabs: disabled
  MessageFormat: disabled
  RussianNovel: disabled
  SingleLineSubject: enabled
```

**Comment by sds** [comment]
Hey @obromios, I'm not understanding how you're seeing the behavior you describe. There are a couple of things that don't make sense to me:

* When you run `overcommit -u`, Overcommit is clearly installed. There shouldn't be a way for you to see the error message about the gem not being installed, unless you explicitly uninstalled it after uninstalling the hooks

* When I try reproducing this locally (installing hooks, then uninstalling) everything works as expected

As for `--list-hooks`, please ignore it. It will always show what hooks _would_ run if Overcommit were to be installed, and does not tell you whether or not the hooks are actually installed.

I'm afraid I can't be of help here. There's something odd with your installation and I don't have suggestions as to what it could be.

**Comment by obromios** [comment]
Thank you, it was helpful to know that it was something odd in installation rather than me doing something odd with overcommit.

The installation was the result of copying the entire disk from an old machine to a new machine. Perhaps this somehow corrupted the overcommit installation.   What I ended up doing was cloning a new copy from the master at Github. This exhibited none of the problems that I have described.  I then manually edited the clone branch to add changes from the older corrupted branch.

One thing that I did find strange is that when I cloned the master, if I did ```gem list overcommit``` it returned
```
*** LOCAL GEMS ***

overcommit (0.28.0)
```
despite the fact that I did not have overcommit in my Gemfile and I had not installed overcommit into the new respository.

Perhaps the problem was to do with version 0.28.0 of overcommit.
